From 7ed210437efe4da6a7c1ef3adf951baf272b01d7 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Thu, 10 Nov 2022 21:54:24 +0000
Subject: [PATCH] DesktopNativeWidgetAura: don't apply DIP for SetBounds

Also, don't scale Display bounds, only its content
should be scaled

Signed-off-by: Martin Jansa <martin.jansa@gmail.com>
---
Upstream-Status: Pending

 src/ui/display/display.cc                                      | 2 +-
 src/ui/views/widget/desktop_aura/desktop_native_widget_aura.cc | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/ui/display/display.cc b/src/ui/display/display.cc
index 088ec6ef7e..0275fc0802 100644
--- a/src/ui/display/display.cc
+++ b/src/ui/display/display.cc
@@ -266,7 +266,7 @@ void Display::SetScaleAndBounds(float device_scale_factor,
   SetScale(device_scale_factor);
 
   gfx::RectF f(bounds_in_pixel);
-  f.Scale(1.f / device_scale_factor_);
+  //f.Scale(1.f / device_scale_factor_);
   bounds_ = gfx::ToEnclosedRectIgnoringError(f, kDisplaySizeAllowanceEpsilon);
   size_in_pixels_ = bounds_in_pixel.size();
   UpdateWorkAreaFromInsets(insets);
diff --git a/src/ui/views/widget/desktop_aura/desktop_native_widget_aura.cc b/src/ui/views/widget/desktop_aura/desktop_native_widget_aura.cc
index 2784ca8a0e..fd081e0958 100644
--- a/src/ui/views/widget/desktop_aura/desktop_native_widget_aura.cc
+++ b/src/ui/views/widget/desktop_aura/desktop_native_widget_aura.cc
@@ -860,7 +860,7 @@ std::string DesktopNativeWidgetAura::GetWorkspace() const {
 void DesktopNativeWidgetAura::SetBounds(const gfx::Rect& bounds) {
   if (!desktop_window_tree_host_)
     return;
-  desktop_window_tree_host_->SetBoundsInDIP(bounds);
+  desktop_window_tree_host_->AsWindowTreeHost()->SetBoundsInPixels(bounds);
 }
 
 void DesktopNativeWidgetAura::SetBoundsConstrained(const gfx::Rect& bounds) {
